lib/optimizely/event/forwarding_event_processor.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+require_relative 'event_processor'
+module Optimizely
+  class ForwardingEventProcessor < EventProcessor
+    # ForwardingEventProcessor is a basic transformation stage for converting
+    # the event batch into a LogEvent to be dispatched.
+    def initialize(event_dispatcher, logger = nil)
+      @event_dispatcher = event_dispatcher
+      @logger = logger || NoOpLogger.new
+    end
+
+    def process(user_event)
+      log_event = Optimizely::EventFactory.create_log_event(user_event, @logger)
+
```

Comment:
Is the log event notification implemented yet? I think it should be triggered here.

Reply:
Implemented in PR: https://github.com/optimizely/ruby-sdk/pull/196